### ISSUE TITLE
Sub-task triggers parent task

### ISSUE THREAD

**Issue by art-in** [betsol/gulp-require-tasks#23]
```
tasks/
    build/
        index.js
        watch/
            index.js
```

Tasks: `build`, `build:watch`  

When running `build:watch` I'm not expecting `build` to run.  
But they both run in parallel.
 
Also if I remove `build/index.js` there's an error: `Task 'build' is not in your gulpfile`  

I think sub-tasks should be independent from parent tasks,  
and users should _explicitly_ run both if they want, instead of _implicit_ run in parallel.    
Eg. user may want to run parent-child in sequence, or do not run parent at all (my case).

In my case both `build` and `build:watch` tasks share same `build/` folder,  
but `build:watch` is not _extension_ of the `build`, but rather - _separate version_ of it.

What do you think? Is that conscious behavior or just a bug?

**Comment by art-in** [comment]
Looks line nobody cares...

**Comment by art-in** [comment]
Turns out I was stupid. My `watch` task was triggering another task, which in turn was triggering `build`.  
So I guess everything is ok here. Closing.
